README.md

```diff
@@ -1,79 +1,250 @@
+
+#### Digital Signature
+
+The following digital signature algorithms from liboqs are supported (assuming they have been enabled in liboqs). Note that only L1 signature and all **Rainbow** variants are enabled by default, and should you wish to enable additional variants, consult [the "Code Generation" section of the documentation in the wiki](https://github.com/open-quantum-safe/openssh/wiki/Using-liboqs-supported-algorithms-in-the-for://github.com/open-quantum-safe/openssh/wiki/Using-liboqs-supported-algorithms-in-the-fork#code-generation).
+
+- `oqsdefault` (see [here](https://github.com/open-quantum-safe/openssh-portable/wiki/Using-liboqs-supported-algorithms-in-the-fork) for what this denotes)
+<!--- OQS_TEMPLATE_FRAGMENT_LIST_ALL_SIGS_START -->
+- **Dilithium**: `dilithium2`, `dilithium3`, `dilithium5`
+<!--- OQS_TEMPLATE_FRAGMENT_LIST_ALL_SIGS_END -->
+
+The following hybrid algorithms are supported; they combine a quantum-safe algorithm listed above with a traditional digital signature algorithm (`<SIG>` is any one of the algorithms listed above):
+
+- if `<SIG>` has L1 security, then the fork provides the methods `rsa3072-<SIG>` and `ecdsa-nistp256-<SIG>`, which combine `<SIG>` with RSA3072 and with ECDSA using NIST's P256 curve respectively.
+- if `<SIG>` has L3 security, the fork provides the method `ecdsa-nistp384-<SIG>`, which combines `<SIG>` with ECDSA using NIST's P384 curve.
+- if `<SIG>` has L5 security, the fork provides the method `ecdsa-nistp521-<SIG>`, which combines `<SIG>` with ECDSA using NIST's P521 curve.
+
+## Quickstart
+
+The steps below have been confirmed to work on macOS 10.14 (clang 10.0.0) and Ubuntu 18.04.1 Bionic (gcc-7).
+
+### Building OQS-OpenSSH
+
+### Step 0: Install dependencies
+
+On **Ubuntu**, you need to install the following packages:
+
+	sudo apt install autoconf automake cmake gcc libtool libssl-dev make ninja-build zlib1g-dev
+
+On **Linux**, you also may need to do the following:
+
+- You may need to create the privilege separation directory:
+
+		sudo mkdir -p -m 0755 /var/empty
+
+- You may need to create the privilege separation user:
+
+		sudo groupadd sshd
+		sudo useradd -g sshd -c 'sshd privsep' -d /var/empty -s /bin/false sshd
+
+On **macOS**, you need to install the following packages using brew (or a package manager of your choice):
+
+	brew install autoconf automake cmake libtool ninja openssl@1.1 wget
+
+### Step 1: Build and install liboqs
+
+The following instructions install liboqs into a subdirectory inside the OpenSSH source. If `<OPENSSH_ROOT>` is the root of the OpenSSH source:
 
 ```
-tar zxvf openssh-X.YpZ.tar.gz
-cd openssh
-./configure # [options]
-make && make tests
+git clone --branch master --single-branch --depth 1 https://github.com/open-quantum-safe/liboqs.git
+cd liboqs
+mkdir build && cd build
+cmake .. -GNinja -DENABLE_SIG_PICNIC=OFF -DCMAKE_POSITION_INDEPENDENT_CODE=ON -DCMAKE_INSTALL_PREFIX=<OPENSSH_ROOT>/oqs
```

Comment:
Ah, this should be `OQS_ENABLE_SIG_PICNIC`, will fix. And it's disabled because on macOS, there's a duplicate symbol error due to `explicit_bzero` being in Picnic's compat.c and in OpenSSH. I'm sure the fix isn't too hard though, just haven't had the time myself.